Update settings for `.resx` files in `.editorconfig`

xtqqczze opened the pull request:
Context: https://github.com/dotnet/runtime/pull/119302#issuecomment-3248692990, https://github.com/dotnet/android/pull/4954.

Also convert `.resx` files to utf8 encoding (remove BOM).

cc: @MichalStrehovsky

xtqqczze committed:
Update `.editorconfig` to adjust settings for `.resx` files to match VS

Context: https://github.com/dotnet/runtime/pull/119302#issuecomment-3248692990, https://github.com/dotnet/android/pull/4954.

tannergooding commented on .editorconfig:
```diff
@@ -176,9 +176,17 @@ charset = utf-8
 
+# Resources - match defaults for VS
+[*.resx]
+charset = utf-8-bom
+indent_size = 2
+indent_style = space
+insert_final_newline = false
+trim_trailing_whitespace = false
```
`indent_style = space` is redundant

`trim_trailing_whitespace` notably matters for resource strings with newlines in them. While we should typically not have those in the first place since that can be non-deterministic or potentially problematic across platforms, it isn't respected by VS today and so we should be explicit and likely log an issue around that.

VS doesn't respect `insert_final_newline = true` either, so we do probably want to keep it explicitly as `false`. A bug should likely be logged for it as well

For `charset`, we do want `utf-8` to ensure that all characters are encoded correctly. However, we've typically made an explicit choice to not use the `bom` (and there's been several discussions around it). VS respects what we set in the editorconfig here, so we should probably just use `utf-8`

xtqqczze replied:
> For `charset`, we do want `utf-8` to ensure that all characters are encoded correctly. However, we've typically made an explicit choice to not use the `bom` (and there's been several discussions around it). VS respects what we set in the editorconfig here, so we should probably just use `utf-8`

A scan using Python’s chardet library shows that the majority of existing files are actually `utf-8-bom` rather than `utf-8`:

```text
Count Name
----- ----
   32 ascii
  230 UTF-8-SIG
```

tannergooding replied:
The existing resx files notably have the BOM because VS defaults to that if editorconfig hasn't overridden it. However, since VS does respect the encoding specified, we can be more consistent with our decision if we specify it explicitly

xtqqczze replied:
Addressed 358720c8684fb9f78bf8c7bf0570cc26b5003e19

tannergooding replied:
VS Feedback: <https://learn.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2022>

xtqqczze committed:
Convert `.resx` files to `utf8` encoding (remove BOM)

The following script was used:
```sh
for file in $(find . -name '*.resx' -type f); do
  gsed -i '1s/^\xEF\xBB\xBF//' $file
done
```
